src/ASM/LR/ASMu2D.C

```diff
@@ -635,20 +635,60 @@ void ASMu2D::constrainEdge (int dir, bool open, int dof, int code, char basis)
   lr->getEdgeFunctions(edgeFunctions, edge);
   lr->getEdgeElements( edgeElements,  edge);
 
+  // find the corners since these are not to be included in the L2-fitting
+  // of the inhomogenuous dirichlet boundaries; corners are interpolatory
+  // Optimization note: loop over the "edge"-conatiner to manually pick up the
+  //                    end nodes. LRspine::getEdgeFunctions() does a global search.
+  std::vector<LR::Basisfunction*> c1, c2;
+  switch (edge)
+  {
+  case LR::SOUTH:
+    lr->getEdgeFunctions(c1, LR::SOUTH_WEST);
+    lr->getEdgeFunctions(c2, LR::SOUTH_EAST);
+    break;
+  case LR::WEST:
+    lr->getEdgeFunctions(c1, LR::SOUTH_WEST);
+    lr->getEdgeFunctions(c2, LR::NORTH_WEST);
+    break;
+  case LR::EAST:
+    lr->getEdgeFunctions(c1, LR::NORTH_EAST);
+    lr->getEdgeFunctions(c2, LR::SOUTH_EAST);
+    break;
+  case LR::NORTH:
+    lr->getEdgeFunctions(c1, LR::NORTH_WEST);
+    lr->getEdgeFunctions(c2, LR::NORTH_EAST);
+    break;
+  default: return;
+  }
+
   // build up the local element/node correspondence neede by the projection
   // call on this edge by ASMu2D::updateDirichlet()
   DirichletEdge de(edgeFunctions.size(), edgeElements.size(), dof, code);
+  de.corners[0] = c1[0]->getId();
+  de.corners[1] = c2[0]->getId();
```

Comment:
look at the old code - i didn't do the getCorner shenanigans because i wanted to complicate things.